### PR TITLE
Add Windows CI system

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR adds the support for appveyor.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - md build
+  - cd build
+  - cmake ..
+  - cmake --build . --config Release --target INSTALL
+  # compile osqp-eigen
+  - cd c:\projects\osqp-eigen
+  - md build
+  - cd build
+  - cmake .. -DEIGEN3_INCLUDE_DIR="C:/Program Files (x86)/Eigen/include/eigen3"
+  - msbuild /m /p:Configuration=Release /p:Platform="Win32" OsqpEigen.sln
```
Comment (traversaro) [inline appveyor.yml]: I guess this line is a duplicate of the following one, so it can be removed.

Comment (GiulioRomualdi) [reply appveyor.yml]: Actually also 
```
  - msbuild /m /p:Configuration=Release /p:Platform="Win32" osqp.sln
  - cmake --build . --config Release
  - cmake --build . --config Release --target INSTALL
```
are duplicated lines. So at the end I can leave only 
```
- cmake --build . --config Release --target INSTALL
```
@traversaro correct me if I'm wrong

Comment (traversaro) [reply appveyor.yml]: Indeed!

Comment (GiulioRomualdi) [reply appveyor.yml]: Ok, so I think that there are duplicated line also [here](https://github.com/robotology/idyntree/blob/master/appveyor.yml).

Comment (traversaro) [reply appveyor.yml]: Good catch!

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+install:
+  # Check env variables
+  - cmd: echo CMAKE_PREFIX_PATH %CMAKE_PREFIX_PATH%
+  - cmd: echo PATH %PATH%
+build:
+
+build_script:
+  # download and build osqp
+  - cd c:\projects
+  - git clone --recursive https://github.com/oxfordcontrol/osqp.git
```
Comment (diegoferigo) [inline appveyor.yml]: You can safely use `--depth 1` in order to clone only the tree of the last commit

**Comment by traversaro** [comment]
Sorry for the delay, approved.

File: appveyor.yml (hunk 2 above)
Comment (GiulioRomualdi) [reply appveyor.yml]: Done in dbda925

File: appveyor.yml (hunk 1 above)
Comment (GiulioRomualdi) [reply appveyor.yml]: Done in dbda925

**Pull request merged**